src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -371,8 +431,25 @@ protected Iterator<Path> pathIteratorForPublisher(
      */
     @Override
     public void createDirectory(Path dir, FileAttribute<?>... attrs) throws IOException {
+        S3FileSystem fs = ((S3Path)dir).getFileSystem();
         try {
-            this.createDirectory(null, dir, attrs);
+            String directoryKey = ((S3Path)dir).toRealPath(NOFOLLOW_LINKS).getKey();
+            if (!directoryKey.endsWith(S3Path.PATH_SEPARATOR) && !directoryKey.isEmpty()) {
```

Comment:
It is like in the original code, but you are right. I have created #134 so that we can fix it after the merge.

https://github.com/awslabs/aws-java-nio-spi-for-s3/blob/e831e674425fd62d48ed4464299d5461e1badad3/src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java#L343-L346